### DISCUSSION TITLE
Bitcrush and resample lo-fi audio effect

### DISCUSSION THREAD

**Discussion by elvisish** [godotengine/godot-proposals#5103] [General Discussions]
As far as I know, there's no way to lofi audio as an effect, like being able to adjust the bitrate, samplerate, interpolation, jitter, etc as an audio effect. It could be very interesting for retro games and indeed horror games that rely on these kinds of glitchy sounds. I was hoping there might be a plug-in that was available, but I don't think there is anything yet.

**Comment by Calinou** [comment]
See https://github.com/godotengine/godot-proposals/issues/2298. The consensus is that you can do this yourself in an audio editor. This requires some manual work, but it's much more flexible this way :slightly_smiling_face:

**Reply by elvisish** [reply]
> See https://github.com/godotengine/godot-proposals/issues/2298. The consensus is that you can do this yourself in an audio editor. This requires some manual work, but it's much more flexible this way :slightly_smiling_face: 

Oh that's definitely what I'm trying to avoid, since it would be better for retro games if there was the option to toggle for players who want full quality sound vs. lofi. It would also be better for procedural effects in-game where the bitcrushing could adapt to gameplay that static rendering wouldn't allow.

**Comment by Akosmo** [comment]
It seems bitcrushing has been possible since the addition of the distortion effect in the engine. You should use `MODE_LOFI`. All it does it reducing bit depth. AFAIK, there's not built-in way to easily downsample audio as an effect.

**Reply by Akosmo** [reply]
[PR for downsampling](https://github.com/godotengine/godot/pull/68768)
